identity-did/src/verification/method_scope.rs

```diff
@@ -51,3 +53,15 @@ impl FromStr for MethodScope {
     }
   }
 }
+
+impl From<MethodRelationship> for MethodScope {
+  fn from(relationship: MethodRelationship) -> Self {
+    match relationship {
+      MethodRelationship::Authentication => Self::Authentication,
+      MethodRelationship::AssertionMethod => Self::AssertionMethod,
+      MethodRelationship::KeyAgreement => Self::KeyAgreement,
+      MethodRelationship::CapabilityDelegation => Self::CapabilityDelegation,
+      MethodRelationship::CapabilityInvocation => Self::CapabilityInvocation,
+    }
+  }
+}
```

Comment:
Ergonomics was the primary reason I copied the variants. I think we don't expect more relationships to be added any time soon, so the maintenance burden of keeping this up-to-date in two places is small.

```rust
// snippet
.create_method()
.scope(MethodScope::VerificationRelationship(MethodRelationship::AssertionMethod))
```

Nested seems acceptable, too. For Wasm in `insert_method`, as an example, we can also parse the string `"AssertionMethod"` into the nested variant, so ergonomics wouldn't be affected. Though, in the future, we might want to rethink whether to take a proper `WasmMethodScope` type there?

Reply:
>  Though, in the future, we might want to rethink whether to take a proper WasmMethodScope type there?

Although I much prefer the nested version of `MethodScope`, since wasm_bindgen only supports flat enums we might need to flatten it manually for the Wasm bindings anyway.

I would be in favour of nesting it in Rust since that seems to make the most sense, and flattening it in Wasm when we implement `WasmMethodScope` (doesn't have to be in this PR).

E.g.
In Rust:
```rust
pub enum MethodScope {
  VerificationMethod,
  VerificationRelationship(MethodRelationship),
}
```
In Wasm bindings:
```rust
pub enum WasmMethodScope {
  VerificationMethod,
  Authentication,
  AssertionMethod,
  KeyAgreement,
  CapabilityDelegation,
  CapabilityInvocation,
}
```
with appropriate `From` implementations for conversions. 

One could also just wrap the `MethodScope` instead, which is how some other enums have been handled.
E.g.
```rust
pub struct WasmMethodScope(MethodScope);
```
Then expose the variants as functions.

Reply:
Nested it on the Rust side now. Wasm still works with strings for now. I think that's a good upgrade to do when getting the Wasm bindings ready for 1.0.